sdk/elasticsan/Azure.ResourceManager.ElasticSan/src/Generated/ElasticSanSnapshotData.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using Azure.Core;
+using Azure.ResourceManager.ElasticSan.Models;
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.ElasticSan
+{
+    /// <summary>
+    /// A class representing the ElasticSanSnapshot data model.
+    /// Response for Volume Snapshot request.
+    /// </summary>
+    public partial class ElasticSanSnapshotData : ResourceData
+    {
+        /// <summary> Initializes a new instance of ElasticSanSnapshotData. </summary>
+        /// <param name="creationData"> Data used when creating a volume snapshot. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="creationData"/> is null. </exception>
+        public ElasticSanSnapshotData(SnapshotCreationData creationData)
+        {
+            Argument.AssertNotNull(creationData, nameof(creationData));
+
+            CreationData = creationData;
+        }
+
+        /// <summary> Initializes a new instance of ElasticSanSnapshotData. </summary>
+        /// <param name="id"> The id. </param>
+        /// <param name="name"> The name. </param>
+        /// <param name="resourceType"> The resourceType. </param>
+        /// <param name="systemData"> The systemData. </param>
+        /// <param name="creationData"> Data used when creating a volume snapshot. </param>
+        /// <param name="provisioningState"> State of the operation on the resource. </param>
+        /// <param name="sourceVolumeSizeGiB"> Size of Source Volume. </param>
+        /// <param name="volumeName"> Source Volume Name of a snapshot. </param>
+        internal ElasticSanSnapshotData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, SnapshotCreationData creationData, ElasticSanProvisioningState? provisioningState, long? sourceVolumeSizeGiB, string volumeName) : base(id, name, resourceType, systemData)
+        {
+            CreationData = creationData;
+            ProvisioningState = provisioningState;
+            SourceVolumeSizeGiB = sourceVolumeSizeGiB;
+            VolumeName = volumeName;
+        }
+
+        /// <summary> Data used when creating a volume snapshot. </summary>
+        internal SnapshotCreationData CreationData { get; set; }
+        /// <summary> Fully qualified resource ID of the volume. E.g. "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ElasticSan/elasticSans/{elasticSanName}/volumegroups/{volumeGroupName}/volumes/{volumeName}". </summary>
+        public string CreationDataSourceId
```

Comment:
There are several `XXXId` properties added. What's the formats of them?

- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

Reply:
Updated to resource identifier.